### PR TITLE
chore(295): add support for mdx stories

### PR DESCRIPTION
**Pull request by jimmynotjim**
Adds support for the newer mdx story format

Screen Shot 2021-04-19 at 2 53 12 PM

### CODE REVIEW

**Comment by twilliams15** [comment]
Saw this PR and wanted to see if we can get ["controls" (or "essentials")](https://storybook.js.org/docs/react/essentials/introduction) added while we're at it?

Might need to update `main.js` to take advantage after installing

Also I'd like to help test this out, but can't figure out how to point to the commit, updating the package.json to use
```"@pixeloven/cli-addon-storybook": "pixeloven/pixeloven#mdx-stories"```
didn't work for me, how do I do it?

File: packages/examples/component-library/package.json [hunk 1]
```diff
@@ -29,6 +29,7 @@
     "@pixeloven/cli": "^7.0.0-alpha.0",
     "@pixeloven/cli-addon-storybook": "^7.0.0-alpha.1",
     "@pixeloven/cli-core": "^7.0.0-alpha.0",
+    "@storybook/addon-docs": "6.2.7",
```
Comment (jimmynotjim) [inline packages/examples/component-library/package.json:32]: Whoops, pretty sure I don't need this, but it was complaining it didn't exist. Lemme remove it and see

File: packages/pixeloven-storybook/common/package.json [hunk 2]
```diff
@@ -19,12 +19,12 @@
...
     "dependencies": {
         "@babel/core": "7.13.8",
-        "@storybook/addon-actions": "6.1.20",
-        "@storybook/addon-backgrounds": "6.1.20",
-        "@storybook/addon-knobs": "6.1.20",
-        "@storybook/addon-viewport": "6.1.20",
-        "@storybook/react": "6.1.20",
-        "@storybook/theming": "6.1.20",
+        "@storybook/addon-actions": "6.2.7",
+        "@storybook/addon-backgrounds": "6.2.7",
```
Comment (jimmynotjim) [inline packages/pixeloven-storybook/common/package.json:23]: Shoot, should have removed the old addons and added essentials here

File: packages/pixeloven-storybook/config/package.json [hunk 3]
```diff
@@ -22,16 +22,13 @@
...
-        "@storybook/addon-docs": "6.1.20",
-        "@storybook/addon-knobs": "6.1.20",
-        "@storybook/addon-viewport": "6.1.20",
-        "@storybook/addons": "6.1.20",
-        "@storybook/core": "6.1.20",
-        "@storybook/react": "6.1.20",
-        "@storybook/theming": "6.1.20",
+        "@storybook/addon-a11y": "6.2.7",
+        "@storybook/addon-docs": "6.2.7",
+        "@storybook/addon-essentials": "6.2.7",
```
Comment (jimmynotjim) [inline packages/pixeloven-storybook/config/package.json:27]: This replaces a bunch of individual addons

File: packages/pixeloven-storybook/config/src/main.ts [hunk 4]
```diff
@@ -3,15 +3,12 @@ import getConfig from "./webpack";
...
-        "@storybook/addon-viewport",
 
         // Tabs
-        "@storybook/addon-knobs",
+        "@storybook/addon-essentials",
         "@storybook/addon-a11y",
-        "@storybook/addon-actions",
     ],
-    stories: [process.cwd() + "/**/*.stories.[tj]s[x]"],
+    stories: [process.cwd() + "/**/*.stories.@(js|ts|jsx|tsx|mdx)"],
```
Comment (jimmynotjim) [inline packages/pixeloven-storybook/config/src/main.ts:12]: This is the new recommended format from storybook

File: packages/pixeloven-webpack/config/src/helpers/shared.ts [hunk 5]
```diff
@@ -340,7 +340,7 @@ export function getSetup(options: Options) {
                 "@src": resolveSourceRoot(),
             },
             extensions: [".js", ".json", ".jsx", ".mjs", ".ts", ".tsx"],
-            modules: [resolveSourceRoot(), "node_modules"],
+            modules: [resolveSourceRoot(), "node_modules", "stories"],
```
Comment (jimmynotjim) [inline packages/pixeloven-webpack/config/src/helpers/shared.ts:343]: gotta resolve stories, I'm not 100% sure why, just following instructions.

**Comment by jimmynotjim** [comment]
@twilliams15 ok things should be good to go. Instructions for testing

* `yarn install` if you never have, this will install Lerna
* `yarn all:bootstrap` this will install all other package stuff
* `yarn all:compile && yarn all:linker` this will build the packages and symlink them
* `cd packages/examples/component-library && yarn start:story`this will start storybook in the example package, go to the url printed in consol

File: packages/examples/component-library/package.json (hunk 1 above)
Comment (ductiletoaster) [reply packages/examples/component-library/package.json:32]: PR looks good just want to remove this if we can.

Comment (jimmynotjim) [reply packages/examples/component-library/package.json:32]: Ohhh, I remember why it's necessary now, you have to import the doc components into the MDX stories

```
import { Meta, Story, Canvas, ArgsTable } from '@storybook/addon-docs/blocks';
```

Comment (ductiletoaster) [inline packages/examples/component-library/package.json:32]: @jimmynotjim This was one of the reason I had the common library so we can wrap dependencies like this.

**Pull request closed without merging**